src/components/floatingButton/index.tsx

```diff
@@ -117,6 +113,9 @@ class FloatingButton extends PureComponent<FloatingButtonProps> {
     const {withoutAnimation, secondaryButton, visible} = this.props;
     const Container = !withoutAnimation ? AnimatableView : View;
 
+    // NOTE: keep this.firstLoad as true as long as the visibility changed to true
+    this.firstLoad && !visible ? this.firstLoad = true : this.firstLoad = false;
```

Comment:
What if it visible=true, then visible=false, and then visible=false again?

Reply:
It won't cause a re-render since the props are the same..
The problem was only when the user sends visible=false at the beginning, and we changed the firstLoad to false even though the component isn't loaded yet